base/error.go

```diff
@@ -237,6 +237,22 @@ func IsDocNotFoundError(err error) bool {
 	}
 }
 
+func IsTimeoutError(err error) bool {
```

Comment:
I think we want to include `gocb.ErrOverload` and `gocb.ErrTemporaryFailure` and `gocb.ErrCircuitBreakerOpen`. This might make the function `IsTemporaryKvError()` 

I wonder if we want to scope this message to just kv write error errors. I'd be curious if `ErrRateLimitedFailure` or `ErrQuotaLimitedFailure` could occur from kv.